packages/react/src/components/Tooltip/next/TooltipDefinition.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { settings } from 'carbon-components';
+import setupGetInstanceId from '../../../tools/setupGetInstanceId';
+import debounce from 'lodash.debounce';
+import { composeEventHandlers } from '../../../tools/events';
+import { keys, matches } from '../../../internal/keyboard';
+
+const { prefix } = settings;
+const getInstanceId = setupGetInstanceId();
+const TooltipDefinition = ({
+  align,
+  children,
+  className,
+  direction,
+  id,
+  content,
+  triggerClassName,
+  onFocus,
+  onBlur,
+  onMouseEnter,
+  onMouseLeave,
+  ...rest
+}) => {
+  const [allowTooltipVisibility, setAllowTooltipVisibility] = useState(true);
+  const [tooltipVisible, setTooltipVisible] = useState(false);
+  const tooltipId = id || `definition-tooltip-${getInstanceId()}`;
+
+  const tooltipClassName = cx(
+    `${prefix}--tooltip--definition`,
+    `${prefix}--tooltip--a11y`,
+    className
+  );
+  const tooltipTriggerClasses = cx(
+    `${prefix}--tooltip__trigger`,
+    `${prefix}--tooltip--a11y`,
+    `${prefix}--tooltip__trigger--definition`,
+    triggerClassName,
+    {
+      [`${prefix}--tooltip--${direction}`]: direction,
+      [`${prefix}--tooltip--align-${align}`]: align,
+      [`${prefix}--tooltip--hidden`]: !allowTooltipVisibility,
+      [`${prefix}--tooltip--visible`]: tooltipVisible,
+    }
+  );
+
+  const debounceTooltipVisible = debounce(() => setTooltipVisible(false), 100);
+
+  const handleFocus = () => {
+    debounceTooltipVisible.cancel();
+    setAllowTooltipVisibility(true);
+    setTooltipVisible(true);
+  };
+
+  const handleBlur = debounceTooltipVisible;
+
+  const handleMouseEnter = () => {
+    debounceTooltipVisible.cancel();
+    setAllowTooltipVisibility(true);
+    setTooltipVisible(true);
+  };
+
+  const handleMouseLeave = debounceTooltipVisible;
+
+  useEffect(() => {
+    const handleEscKeyDown = (event) => {
+      if (matches(event, [keys.Escape])) {
+        setAllowTooltipVisibility(false);
+      }
+    };
+    document.addEventListener('keydown', handleEscKeyDown);
+    return () => document.removeEventListener('keydown', handleEscKeyDown);
+  }, []);
+
+  return (
+    <div
+      {...rest}
+      className={tooltipClassName}
+      onMouseEnter={composeEventHandlers([onMouseEnter, handleMouseEnter])}
+      onMouseLeave={composeEventHandlers([onMouseLeave, handleMouseLeave])}>
+      <button
+        type="button"
+        className={tooltipTriggerClasses}
+        aria-describedby={tooltipId}
+        onFocus={composeEventHandlers([onFocus, handleFocus])}
+        onBlur={composeEventHandlers([onBlur, handleBlur])}>
+        {children}
+      </button>
+      <div
```

Comment:
We should make sure this component uses `Popover` directly, it should mean that we won't need the debounce or mouseover/mouseout handlers 👍